src/main/java/uk/gov/companieshouse/mapper/CompanyOfficerMapper.java

```diff
@@ -0,0 +1,46 @@
+package uk.gov.companieshouse.mapper;
+
+import org.springframework.stereotype.Service;
+import uk.gov.companieshouse.api.model.officers.CompanyOfficerApi;
+import uk.gov.companieshouse.model.db.dissolution.DissolutionDirector;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class CompanyOfficerMapper {
+    public boolean mapCompanyOfficersToIsOverHalfSelected(
+            List<CompanyOfficerApi> companyOfficers,
+            List<DissolutionDirector> selectedDirectors
+            ) {
+
+        final List<String> activeOfficers = this.mapCompanyOfficersToActiveList(companyOfficers);
+        final List<String> selectedOfficers = this.mapDissolutionDirectorsToSelectedOfficers(selectedDirectors);
+
+        final float officersSelected = selectedOfficers.stream().filter(
```

Comment:
What if an officer is provided that does not exist?
What if the same officer is provided multiple times?

I would expect an error message to be returned for each of these scenarios. This ties into the comment about having a dissolutionValidator which is capable of flagging what was the cause of the issue.